dart-common/src/main/java/com/f2prateek/dart/common/AbstractDartProcessor.java

```diff
@@ -134,19 +117,23 @@ private void parseInjectExtraAnnotatedElements(RoundEnvironment env,
     }
   }
 
-  private void parseHensonAnnotatedElements(RoundEnvironment env,
+  protected void parseHensonNavigableAnnotatedElements(RoundEnvironment env,
       Map<TypeElement, InjectionTarget> targetClassMap, Set<TypeMirror> erasedTargetTypes) {
-    for (Element element : env.getElementsAnnotatedWith(Henson.class)) {
+    List<TypeElement> modelTypeElements = new ArrayList<>();
+    for (Element element : env.getElementsAnnotatedWith(HensonNavigable.class)) {
       try {
-        parseHenson((TypeElement) element, targetClassMap, erasedTargetTypes);
+        parseHenson((TypeElement) element, targetClassMap, erasedTargetTypes, modelTypeElements);
       } catch (Exception e) {
         StringWriter stackTrace = new StringWriter();
         e.printStackTrace(new PrintWriter(stackTrace));
 
-        error(element, "Unable to generate extra injector when parsing @Henson.\n\n%s",
+        error(element, "Unable to generate extra injector when parsing @HensonNavigable.\n\n%s",
             stackTrace.toString());
       }
     }
+    for (TypeElement modelTypeElement : modelTypeElements) {
```

Comment:
we remove the model classes from the list of injection targets. We don't want to generate an intent builder for them.